### DISCUSSION TITLE
ASB log is confusing

### DISCUSSION THREAD

**Discussion by ghost** [comit-network/xmr-btc-swap#760] [Q&A] [answered]
Hey guys
Why messages are not timestamped? 
Is the log file rewritten every time `asb` is restarted?  I've had restarted `asb` a few times before, but didn't notice anything and thought new messages get appended to it. Last time I restarted it a couple of hours ago; I just checked the log file and noticed some messages have disappeared, scrolled to the top and it seems to be rewritten. Although there are hundreds of "new connection/lost connection" messages which I doubt they are all for the past two hours.
Any idea what's happened?

**Comment by rishflab** [comment, marked as answer]
I just had a look at the code for the ASB and we don't do any logging to a file. It must be something to do with how/where you are piping the asb output. How are you running the ASB?

**Reply by ghost** [reply]
I followed @sethsimmons'  [guide](https://sethforprivacy.com/guides/run-an-atomic-swap-provider-advanced/)  which runs `asb` through sysetmd and logs the output to a file. 
So,  nothing's wrong with `asb` here.  Sorry for the trouble, and **thank you so much** for clarifying the issue.

**Comment by thomaseizinger** [comment]
Messages are only timestamped if you run `asb` in an interactive terminal: https://github.com/comit-network/xmr-btc-swap/blob/413a2d37aa4c2bb950689a1b90532b2a4239edd8/swap/src/asb/tracing.rs#L22-L26

If you pipe the log messages to something else, that tool is expected to capture the timestamp. For example, if you run `asb` through systemd, `journalctl` will timestamp the log messages.

**Reply by thomaseizinger** [reply]
@rishflab We are probably trying to be too clever here and should just expose a switch on whether or not we want timestamps in the logs and let the user decide.

**Reply by rishflab** [reply]
alright, should be a quick fix.

**Reply by rishflab** [reply]
Should the timestamp flag apply to the json output. It seems inconsistent if it didn't but if we do, we introduce a breaking change for people parsing json.
